docs/book/src/development/conventions.md

```diff
@@ -36,3 +36,21 @@ use
 ```
 
 And then within the code you can check the length or range over the slice.
+
+## Tests
+
+There are two types of tests written for CAPA controllers in this repo:
+* Unit tests  
+* Integration tests
+
+If any new unit or integration tests has to be added in this repo,we should follow the below conventions.
+
+### Unit tests
+These tests are written to test the functions inside the same controller file where we perform sanity checks, functionality checks etc. Here we use all types of mocking like [fakeclient](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake) to mock the services API calls. 
+
+These tests go into the file with suffix *_unit_test.go.
```

Comment:
I don't think we have the test file names the right way around. Personally, `_test.go` should be unit tests and some other pattern for integration tests.

Reply:
That's what we have been following from start, but I will let @sedefsavas weigh in here, if renaming has to be done, I will raise another PR for the same, and change these suffix here.

Reply:
Lets chat in office hours :) We can resolve this until then